test/hotspot/jtreg/runtime/Thread/TestAlwaysPreTouchStacks.java

```diff
@@ -103,83 +163,30 @@ public static void main(String[] args) throws Exception {
-                    long committed = Long.parseLong(m.group(2));
-                    System.out.println(">>>>> " + line + ": " + reserved + " - " + committed);
-                    // This is a bit fuzzy: even with PreTouch we don't commit the full range of what NMT counts
-                    // as thread stack. But without pre-touching, the thread stacks would be committed to about 1/5th
-                    // of their reserved size. Requiring them to be committed for over 3/4th shows that pretouch is
-                    // really working.
-                    if (preTouch && (double)committed < ((double)reserved * 0.75)) {
-                        throw new RuntimeException("Expected a higher ratio between stack committed and reserved.");
-                    } else if (!preTouch && (double)committed > ((double)reserved * 0.50)){
-                        throw new RuntimeException("Expected a lower ratio between stack committed and reserved.");
-                    }
-                    // Added sanity tests: we expect our test threads to be still alive when NMT prints its final
-                    // report, so their stacks should dominate the NMT-reported total stack size.
-                    long max_reserved = memoryCeilingMB * 3 * MB;
-                    long min_reserved = memoryCeilingMB * MB;
-                    if (reserved >= max_reserved || reserved < min_reserved) {
-                        throw new RuntimeException("Total reserved stack sizes outside of our expectations (" + reserved +
-                                                   ", expected " + min_reserved + ".." + max_reserved + ")");
-                    }
-                    foundLine = true;
-                    break;
-                }
-            }
-            if (!foundLine) {
-                throw new RuntimeException("Did not find expected NMT output");
-            }
-        }
-
+          ReservedCommitted pretouch_result = runPreTouchTest(true);
+          ReservedCommitted no_pretouch_result = runPreTouchTest(false);
+          if (pretouch_result.reserved == 0 || no_pretouch_result.reserved == 0) {
+            throw new RuntimeException("Could not run with PreTouch flag.");
+          }
+          double ratio_with = ((double)pretouch_result.committed) / pretouch_result.reserved;
+          double ratio_without = ((double)no_pretouch_result.committed) / no_pretouch_result.reserved;
+          System.out.println("ratio with PreTouch: " + ratio_with + " w/out: " + ratio_without);
+          if (ratio_without > 0.50) {
+            throw new RuntimeException("Expected a lower ratio between stack committed and reserved.");
+          }
+          if (ratio_with < ratio_without) {
+            throw new RuntimeException("Expected a higher ratio between stack committed and reserved.");
+          }
+          double estimated_stack_usage = 1 * MB;
+          double expected_stack_use_with_pretouch = 0.75 * threadStackSizeMB * MB;
```

Comment:
Can you hard-code both values up where the other constants are?

I also would name them clearly for what they are. "estimated_stack_usage" sounds wrong, since we don't really estimate that, the real number should be far lower (for no pretouch) and far higher (for pretouch).

Proposal: `max_stack_usage_with_pretouch` and `min_stack_usage_with_pretouch`

Reply:
Done.